zenoh-flow/src/runtime/dataflow/instance/builtin/zenoh.rs

```diff
@@ -268,15 +290,47 @@ impl<'a> Sink for ZenohSink<'a> {
         let mut sink_inputs: HashMap<PortId, InputRaw> = HashMap::new();
         let mut publishers: HashMap<PortId, Publisher<'a>> = HashMap::new();
 
+        let id = uuid::Uuid::new_v4().to_string();
         match configuration {
             Some(configuration) => {
-                let configuration = configuration.as_object().ok_or(zferror!(
+                let shm_elem_size = configuration
+                    .get(KEY_SHM_ELEM_SIZE)
+                    .unwrap_or(&serde_json::Value::from(
+                        *context.shared_memory_element_size() as u64,
+                    ))
+                    .as_u64()
+                    .unwrap_or(*context.shared_memory_element_size() as u64)
+                    as usize;
+
+                let shm_elem_count = configuration
+                    .get(KEY_SHM_TOTAL_ELEMENTS)
+                    .unwrap_or(&serde_json::Value::from(
+                        *context.shared_memory_elements() as u64
+                    ))
+                    .as_u64()
+                    .unwrap_or(*context.shared_memory_elements() as u64)
+                    as usize;
+
+                let shm_size = shm_elem_size * shm_elem_count;
+
+                let shm_backoff = configuration
+                    .get(KEY_SHM_BACKOFF)
+                    .unwrap_or(&serde_json::Value::from(*context.shared_memory_backoff()))
+                    .as_u64()
+                    .unwrap_or(*context.shared_memory_backoff());
```

Comment:
We can refactor this with a closure (I'm not sure it's a 100% correct Rust code):

```rust
let get_or_default = |configuration, key, default| {
    if let Some(value) = configuration.get(key) {
        if let Ok(value) = value.as_u64() {
            return value;
        }
        log::warn!("Failed to parse / interpret provided value for {key} into u64, using default value: {default}");
    }

    default
};

let shm_elem_size = get_or_default(configuration, KEY_SHM_ELEM_SIZE, context.shared_memory_element_size() as u64) as usize;
let shm_elem_count = get_or_default(configuration, KEY_SHM_TOTAL_ELEMENTS, context.shared_memory_elements() as u64) as usize;
// etc…
```